doc/tutorial/nan_tutorial.txt

```diff
@@ -0,0 +1,65 @@
+
+.. _nan_tutorial:
+
+=================
+Dealing with NaNs
+=================
+
+Having a model yielding NaNs or Infs is quite common if some of the tiny
+components in your model are not set properly. NaNs are hard to deal with
+because sometimes it is caused by a bug or error in the code, sometimes it's
+because of the numerical stability of your actual computing systerm, and even,
+sometimes it relates to your algorithm. Here we try to outline common, basic
+issues which cause the model to yield NaNs, as well as provide nails and
+hammers to diagnose it. 
+
+
+Check Superparameters and Weight Initialization
+-----------------------------------------------
+
+Most frequently, the cause would be that some of the superparameters, especially
```

Comment:
In fact just search and replace superparameter with hyperparameter, that's the more common name